### ISSUE TITLE
b-calendar (and b-form-datepicker) date-disabled-fn should respect min and max restrictions

### ISSUE THREAD

**Issue by mfrieling** [bootstrap-vue/bootstrap-vue#5509]
### Describe the bug

I use the calendar component to let the customer select a date for an appointment request, using a list of available dates provided by the server. To reduce the number of calls to `date-disabled-fn` (which each times must check if the date parameter is in the array of available dates or not) I intend to set the min value to the first date of the array and the max value to the last date of the array (which is sorted correctly).

But it turned out that the component asks for every date which is visible - ignoring min and max - if it should be disabled or not, even when navigating months forward or backward with no date which should be enabled.

### Steps to reproduce the bug

1. Add the `b-calendar` component
2. Set the min value to e. g. the 10th of the current month
3. Set the max value to e. g. the 15th of the current month
4. Add a `date-disabled-fn` function:
```
methods: {
  isDisabled(ymd, date) {
    console.log("Disable date? ", ymd)
    return false
  }
}
```
5. Open the component containing the calendar in the browser with the console window open. You will see the function called 35 times (if the calendar displays 5 full weeks including according dates from previous and next month) instead of only 6 expected calls (10, 11, 12, 13, 15 of current month)
6. Navigate one month forward or backward and again see the function called 35 times instead of not being called at all.

### Expected behavior

* Before calling `date-disabled-fn` check the min and max value, assuming that `date-disabled-fn` will only return false (not disabling a date) which is within the range [min .. max] (inclusive) to prevent unnecessary calls.
* If min is set and `min > date2check` don't call `date-disabled-fn` and disable the date.
* If max is set and `max < date2check` don't call `date-disabled-fn` and disable the date.

### Versions

**Libraries:**

- BootstrapVue: 2.15.0
- Bootstrap: 4.5.0
- Vue: 2.6.11

**Environment:**

- OS: Linux Mint 19.3
- Browser: Chrome
- Version: 83.0.4103.97

**Comment by Hiws** [comment]
Could you create a reproduction of the issue.
Because when i tried to recreate it, it *did* respect min/max values.

<https://codepen.io/Hiws/pen/jOWVwGm>

**Comment by mfrieling** [comment]
OK, just found out that it more or less works.

The problem was that due to get the date selection logic to work with the way, available dates are delivered by the server and providing the callbacks to the calendar, I had to change between date object and ymd format and have overseen to change the `min` and `max` computed functions accordingly.

One caveat I still have is, that it only works if I fetch the available dates during using `created()` lifecycle callback but not if I fetch them during using `activated()` lifecycle callback. As the server call is not done by the component holding the calendar but by the parent component which provides the user a switch between calendar and another mode (3 pre-selected dates) which is the default, `created()` works in this case. But if I had to load the dates from the server first (which for my understanding should be done in `activated()` and not in `created()`) and only can provide correct input values to the calendar after a successful server call, the calendar wouldn't respect them. Here a way to trigger the calendar to update itself with the current inputs could be helpful here.
